.travis.yml

```diff
@@ -551,26 +623,27 @@ matrix:
         - ulimit -n 8192
       script:
         - ./build-support/bin/travis-ci.sh -be
+
     # Rust on macOS
-    - name: "Rust + Platform-specific Tests OSX"
-      os: osx
+    - <<: *default_osx_test_config
+      name: "OSX Rust + platform-specific tests"
+      env:
+        - CACHE_NAME=macosrusttests
       # Fuse actually works on this image. It hangs on many others.
       osx_image: xcode8.3
-      stage: *test
-      language: generic
       before_install:
+        - curl -L https://github.com/stedolan/jq/releases/download/jq-1.5/jq-osx-amd64 -o /usr/local/bin/jq
+        - chmod 755 /usr/local/bin/jq
+        - ./build-support/bin/install_aws_cli_for_ci.sh
         - brew tap caskroom/cask && brew update && brew cask install osxfuse
-      before_script:
-        - ulimit -c unlimited
-        - ulimit -n 8192
       script:
-        # Platform-specific tests currently need a pants pex, so we bootstrap here (no -b) and
-        # set -z to run the platform-specific tests.
-        - ./build-support/bin/travis-ci.sh -ez
+        - ./build-support/bin/travis-ci.sh -bez
 
     # Rust Clippy on Linux
     - <<: *linux_with_fuse
-      name: "Rust Clippy on Linux"
+      name: "Linux Rust Clippy"
+      env:
+        - CACHE_NAME=linuxclippy
```

Comment:
This job should use the `native_engine_cache_config` (and maybe share it with the linux bootstrap shard)

Reply:
Per the travis documentation, sharing caches between jobs is a no-no. Will add the cache config though. 

Is this job recompiling a lot of rust code that the bootstrap job already compiled? Would it make sense to move the clippy run to there?

Reply:
It's re-compiling all of it, but in a way which is different (both cheaper and more expensive) than the bootstrap job. It's doing compiler-integrated linting, which adds extra instrumentation (more expensive), but also doesn't actually do full codegen or linking (cheaper). It probably makes sense to merge it with the rust test running shard, actually...